service/controller/chartconfig/v7/resource/chartstatus/create.go

```diff
@@ -29,30 +31,54 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	if key.ReleaseStatus(customObject) != releaseContent.Status {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, releaseContent.Status))
-
-		customObjectCopy := customObject.DeepCopy()
-		customObjectCopy.Status.ReleaseStatus = releaseContent.Status
+	var status, reason string
+	{
+		if key.IsCordoned(customObject) {
+			status = releaseStatusCordoned
+			reason = key.CordonReason(customObject)
+		} else {
+			status = releaseContent.Status
+			if releaseContent.Status != releaseStatusDeployed {
+				releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
+				if err != nil {
+					return microerror.Mask(err)
+				}
 
-		if releaseContent.Status != releaseStatusDeployed {
-			releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
-			if err != nil {
-				return microerror.Mask(err)
+				reason = releaseHistory.Description
 			}
+		}
+	}
 
-			customObjectCopy.Status.Reason = releaseHistory.Description
-		} else {
-			customObjectCopy.Status.Reason = ""
+	chartConfigStatus := v1alpha1.ChartConfigStatus{
+		ReleaseStatus: status,
+		Reason:        reason,
+	}
+
+	if customObject.Status != chartConfigStatus {
+		// Get chartconfig CR again to ensure the resource version is correct.
+		currentCR, err := r.g8sClient.CoreV1alpha1().ChartConfigs(customObject.Namespace).Get(customObject.Name, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if currentCR.Status.Reason != chartConfigStatus.Reason {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting reason for release '%s' to '%s'", releaseName, reason))
+			currentCR.Status.Reason = chartConfigStatus.Reason
```

Comment:
Why is the log talking about the value of `reason` and the status is actually set by `chartConfigStatus.Reason`? This diverge is the same kind of problem we tried to fix with the update separation of the status fields. Consistency when manipulating and reporting information is very important. Why even have `chartConfigStatus`? The code does some back and forth with the variables even though the mechanism is pretty straight forward. When you miss a clear idea of how this could be simpler, we can pair if you like.